pyfr/backends/openmp/kernels/par-xsmm.mako

```diff
@@ -8,15 +8,18 @@ typedef void (*libxsmm_xfsspmdm_execute)(void *,
 
 void
 par_xsmm(libxsmm_xfsspmdm_execute exec, void *blockk, void *cleank,
-         int n, int nblock,
-         const fpdtype_t *b, fpdtype_t *c)
+         int n, int nblock, const fpdtype_t *b, fpdtype_t *c,
+         int nbcol, int bblocksz, int cblocksz)
 {
-    int endb = n - n % nblock;
-
+    int nci = nbcol / nblock;
+    int rem = nbcol % nblock;
     #pragma omp parallel for
-    for (int i = 0; i < endb; i += nblock)
-        exec(blockk, b + i, c + i);
+    for (int ib = 0; ib < n / nbcol; ib++)
+    {
+        for (int i = 0; i < nci; i++)
+            exec(blockk, b + ib*bblocksz + i*nblock, c + ib*cblocksz + i*nblock);
 
-    if (endb != n)
-        exec(cleank, b + endb, c + endb);
+        if (rem != 0)
+            exec(cleank, b + ib*bblocksz + nci*nblock, c + ib*cblocksz + nci*nblock);
```

Comment:
Is it possible to avoid this?

Reply:
If we generate a single xsmm kernel with N = nbcol then yes we can avoid cleank. And it will be much easier to unify with par-gimmik.mako then.